Remove old mqttAD entities + fixes

lucasnz opened the pull request:
Added a function to remove old/renamed mqttAD entities
Fixed some typos
Renamed some entities to align json format and entity name

wayne-love reviewed (commented):
I'm not a fan the delete old code, whilst it works, it quickly becomes irrelevant.  I'd prefer just to publish instructions for people to delete old entities using something like mqttExplorer so that they can do this as needed.

lucasnz commented:
Fair. I did this as it was causingan issue for you. I could create a button on the website that does this. But I'll leave this off for now. 

I want to tweak the json a bit more too. That might mean more tweaks to the names. So maybe don't merge this yet. I will post a discussion point around this on discord.

lucasnz commented:
I've redone this PR. The first two commits are already in master, so not much to look at there. The last commit has a whole raft of updates. including:
Improve setting discoveryTopic
Renamed some entities to align json format and entity name
Added pump modes to json

I am tweaking the WebUI, but that's not in this commit. Nothing too major, just cleaning things up. I was looking at adding a button to clear the ha auto discovery from mqtt. But that really needs, the mqttHaAutoDiscovery function to move into HAAutoDiscovery.cpp - I think that needs a little thought before it's actioned.

But if you're happy with the updated commit, then it's probably a good point to resync.

wayne-love commented:
We should avoid renaming entities if at all possible. It messes up history in HA. I'll ping you on discord to chat once I'm back online.
